pkg/plugin/v3/scaffolds/internal/templates/config/crd/enablecainjection_patch.go

```diff
@@ -28,6 +28,9 @@ var _ file.Template = &EnableCAInjectionPatch{}
 type EnableCAInjectionPatch struct {
 	file.TemplateMixin
 	file.ResourceMixin
+
+	// Version of CRD patch to generate.
+	CRDVersion string
```

Comment:
```suggestion
	ApiVersion string
```

Reply:
IMO it cannot use the CRD version. By following the suggestion https://github.com/kubernetes-sigs/kubebuilder/pull/1644#discussion_r491227667 it would have its own version stored which means that will allow in the future address to avoid breaking changes and address use cases such as:

- The default version used by the create api and create webhook plugins to scaffold the files ought to be defined in the plugin code, so that, we can have a default version for v2+ and another for v3+ and we will not need a migration guide step to describe to users update their PROJECT file when it changes.
- The design should allow we have a default version for each plugin(create api and create webhook) so that, we will not face problems to address scenarios such as the k8s api deprecate the v1 for webhooks and but not for CRDs.

Reply:
I am closing this one. I think I catcher your idea.